M2/Macaulay2/m2/orderedmonoidrings.m2

```diff
@@ -268,7 +268,21 @@ Ring OrderedMonoid := PolynomialRing => (			  -- no memoize
 		   f=f*RM_(-minexps); -- get rid of monomial in factor if f Laurent polynomial
 		   c=RM_minexps;
 		   );
-	       (facs,exps) := rawFactor raw f;	-- example value: ((11, x+1, x-1, 2x+3), (1, 1, 1, 1)); constant term is first, if there is one
+	       isNumberField := F -> isField F and instance(baseRing F, QuotientRing) and coefficientRing baseRing F === QQ and numgens baseRing F == 1 and numgens ideal baseRing F == 1; 
```

Comment:
This function doesn't always work:
```m2
Macaulay2, version 1.17.2.1
with packages: ConwayPolynomials, Elimination, IntegralClosure, InverseSystems, LLLBases,
               MinimalPrimes, PrimaryDecomposition, ReesAlgebra, Saturation, TangentCone

i1 : isNumberField := F -> isField F and instance(baseRing F, QuotientRing) and coefficientRing baseRing F === QQ and numgens baseRing F == 1 and numgens ideal baseRing F == 1;

i2 : K = toField ( QQ[x,y]/(x^2-3, y^2-5) )

o2 = K

o2 : PolynomialRing

i3 : isNumberField K

o3 = false
```

Reply:
Yes... But to call `rawFactor` we need an extension with only one minimal polynomial.

Reply:
Also there is this function from my package that can help in this case.
```macaulay2
i1 : needsPackage "RationalPoints2"                                                                                                                                                            
                                                                                                                                                                                               
o1 = RationalPoints2                                                                                                                                                                           
                                                                                                                                                                                               
o1 : Package                                                                                                                                                                                   
                                                                                                                                                                                               
i2 : QQ[x]; extField{x^2-3, x^2-5}                                                                                                                                                             
                                                                                                                                                                                               
                QQ[a]                                                                                                                                                                          
o3 = --------------------------[]                                                                                                                                                              
      4     3      2                                                                                                                                                                           
     a  - 2a  - 10a  + 26a - 11                                                                                                                                                                
                                                                                                                                                                                               
o3 : PolynomialRing
```